Implement option to exit after first failure

FrNecas opened the pull request:
This PR introduces a new key and option that stops execution after the first failing test to avoid having to wait for all tests to finish running.

Resolves: https://github.com/psss/tmt/issues/663

lukaszachy commented on tmt/steps/execute/internal.py:
```diff
@@ -259,6 +262,11 @@ def go(self):
                 if self._handle_reboot(test, guest):
                     continue
                 self._results.append(self.check(test))
+                if exit_first and self._results[-1].result == 'fail':
```
How about 'warn' and 'error'? Per docs all three results should be treated as fail (by automation). Also this option is more about 'first test that doesn't pass' than just an explicit fail (IMO).

 Also detached checks for non-zero exit_code which hits timeout (== error)  so this won't be consistent.

FrNecas replied:
Good catch. According to `Result._results`, there can also be `info` (which I have never encountered yet but I guess that can occur). Lets change this condition to `self._results[-1].result not in ('pass', 'info')` then.

psss reviewed (approved):
Looks very good, thanks for implmenting this! Just one nitpick and one test-related question.

FrNecas commented on tests/execute/exit-first/data/plan.fmf:
```diff
@@ -0,0 +1,12 @@
+
+/exit-first:
+    execute+:
+        exit-first+: true
```
Good point, will fix and squash the commits

FrNecas commented:
Both of these should be resolved, thanks for the review. Rebased to master, squashed the commits. I adjusted the test coverage slightly, used the new `--scratch` option instead of `--force` and removed the `-h` as you suggested. In order to test with detach, I had to extend the testing plan a bit for this test case so now all cases should hopefully be covered.

Also just one more small improvement added - rather than using the hardcoded `['exit-first']` list in two places, I added it as `_keys` attribute to `ExecutePlugin`. @psss please take a look at those changes and let me know if there is anything else to do, it should be ready for merging from my point of view :)

psss commented:
Looks great, thanks for all the improvements!
